tensordict/prototype/tensorclass.py

```diff
@@ -532,25 +537,166 @@ def _batch_size_setter(self, new_size: torch.Size) -> None:
     self._tensordict._batch_size_setter(new_size)
 
 
+def _any(self, dim: int = None):
+    """A recursive implementation of `any()` over the tensorclass leaves.
+
+    If the `dim` arg is passed, the resulting tensorclass will have
+    this dimension removed.
+
+    Args:
+        dim (int, optional): if provided, ``any`` will run over that dimension and reduce it.
+            When traversing the tensorclass leaves,  negative dims will be turned to positive
+            ones compared with the root tensorclass' batch-size to avoid any clash
+            of dimensions.
+            Defaults to None, ie full reduction in a single boolean value.
+
+    Examples:
+        >>> @tensorclass
+        ... class MyClass:
+        ...      X: Tensor
+        ...      y: "MyClass"
+        ...
+        >>> c = MyClass(
+        ...     torch.randn(3, 4),
+        ...     MyClass(torch.randn(3, 4, 1), None, batch_size=[3, 4, 1]),
+        ...     batch_size=[3, 4])
+        >>> _ = c.any()  # bool
+        >>> # one can work with negative dimensions, which will be converted to positive dims for the root tensorclass
+        >>> assert (c.any(-1) == c.any(1)).all()
+        >>> assert c.any(-1).y.batch_size == c.any(1).y.batch_size
+
+    """
+    if dim is not None and dim < 0:
+        dim = self.batch_dims + dim
+
+    if dim is not None:
+        non_tensor = {
+            key: None if not is_tensorclass(obj) else obj.any(dim=dim)
+            for key, obj in self._non_tensordict.items()
+        }
+        return self._from_tensordict(
+            self._tensordict.any(dim=dim),
+            non_tensor,
+        )
+    return self._tensordict.any() or any(
+        value.any() for value in self._non_tensordict.values() if is_tensorclass(value)
+    )
+
+
+def _state_dict(self):
+    """Returns a state_dict dictionary that can be used to save and load data from a tensorclass."""
+    state_dict = self._tensordict.state_dict()
+    state_dict["_non_tensordict"] = {
+        key: value if not is_tensorclass(value) else value.state_dict()
+        for key, value in self._non_tensordict.items()
+    }
+    return state_dict
+
+
+def _load_state_dict(self, state_dict):
+    """Loads a state_dict attemptedly in-place on the destination tensorclass."""
```

Comment:
we are not checking that state dict contains all the keys needed

Reply:
imagine you have a "filled" tensorclass. then you load it from a "partial" state dict. what would you expect as a result?
- a raise
- a tensorclass with the "partial" state + others None
- a tensorclass with the "partial" state + others old value

Reply:
To me the latter would make more sense -- at least with the use cases I have in mind.